subprojects/core/src/testFixtures/groovy/org/gradle/util/TestUtil.groovy

```diff
@@ -86,13 +88,16 @@ class TestUtil {
 
     static InstantiatorFactory instantiatorFactory() {
         if (instantiatorFactory == null) {
-            NativeServicesTestFixture.initialize()
-            def annotationHandlers = ProjectBuilderImpl.getGlobalServices().getAll(InjectAnnotationHandler.class)
-            instantiatorFactory = new DefaultInstantiatorFactory(new TestCrossBuildInMemoryCacheFactory(), annotationHandlers, new OutputPropertyRoleAnnotationHandler([]))
+            instantiatorFactory = createInstantiatorFactory({ [] })
```

Comment:
💭 This was changed so we can use `TestUtil.newInstance()` without discovering global services, since  discovering global services requires `distribution-core` on the classpath.

It seems we need to discover `InjectAnnotationHandler` only in `[DefaultVariantTransformRegistryTest.groovy](https://github.com/gradle/gradle/pull/30778/files#diff-2311563e22c51774b43afe7f774f59804a333136a3f3313954db5ce20792b4e4)` test and not everywhere.